spec/models/zone_spec.rb

```diff
@@ -192,6 +192,28 @@
     end
   end
 
+  context "maintenance zone" do
+    it "creates missing region during seed" do
+      expect(MiqRegion.my_region).to be_nil
+      described_class.seed
+      expect(MiqRegion.my_region).to be_present
+    end
+
+    it "is seeded with relation to region" do
+      described_class.seed
+      zone = Zone.where("name LIKE ?", "#{Zone::MAINTENANCE_ZONE_NAME_PREFIX}%").first
+
+      expect(described_class.maintenance_zone).to be_present
```

Comment:
This is unnecessary since the next line will check this and more...

Reply:
next line will check id, which can raise exception instead

Reply:
with comment below, both objects can be nil, which is wrong and your jointed `expects` will pass with this error

Reply:
@slemrmartin How will it pass?  Won't you get an undefined method `id` for NilClass?

Reply:
@Fryguy yes, but with removing this line and remove `.id` from next line there can happen `expect(nil).to eq(nil)`

Reply:
Maybe the expectation should be something like this instead:
```ruby
expect(Zone.maintenance_zone).to have_attributes(
  :name => a_string_starting_with("__maintenance__"),
  ...
)
```

Reply:
Yeah I agree @bdunne that makes more sense